src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -600,6 +602,24 @@ public static void GetFullPath_Windows_StrangeButLegalPaths()
         [InlineData(@"\\.\C:\Foo2\..")]
         public static void GetFullPath_Windows_ValidExtendedPaths(string path)
         {
+            if (PathFeatures.IsUsingLegacyPathNormalization())
+            {
+                // Legacy Path doesn't support any of these paths.
```

Comment:
This could also be added as an extension method 
```cs
AssertExtensions.ThrowsAny<TFirstExceptionType, TSecondExceptionType>(Action action)
   where TFirstExceptionType : Exception
   where TSecondExceptionType : Exception
   {
         try
         {
                ...
         }
         catch (Exception e)
         {
                ...
         }
   }
```

But didn't know if we wanted to add an extension method for one usage, perhaps it could be used in the future.